app/scripts/metamask-controller.js

```diff
@@ -587,13 +589,44 @@ module.exports = class MetamaskController extends EventEmitter {
   // Used when creating a first vault, to allow confirmation.
   // Also used when revealing the seed words in the confirmation view.
   placeSeedWords (cb) {
+
+    this.verifySeedPhrase((err, seedWords) => {
+
+      if (err) {
+        return cb(err)
+      }
+      this.configManager.setSeedWords(seedWords)
+      return cb(null, seedWords)
+    })
+  }
+
+  // Verifies the current vault's seed words if they can restore the
+  // accounts belonging to the current vault.
+  //
+  // Called when the first account is created and on unlocking the vault.
+  verifySeedPhrase (cb) {
```

Comment:
Not a blocker, but a note, we should probably document:

We are trying to use promises for new methods that we add. I know that `placeSeedWords` uses the callback pattern still, but we are moving away from it.  Ideally using the `async/await` pattern. It could really flatten out this code, like this (simplified for example):

```javascript
async verifySeedPhrase () {
  const primaryKeyring = this.keyringController.getKeyringsByType('HD Key Tree')[0]
  const serialized = await primaryKeyring.serialized()
  const seedWords = serialized.mnemonic
  const accounts = await primaryKeyring.getAccounts()
  return seedPhraseVerifier.verifyAccounts(accounts, seedWords)
}
```

Reply:
I was not sure about this, because I saw both patterns in MetamaskController. I'm going to change it to async.